### PR TITLE
#1 Create the esn-frontend-inbox-calendar module

### LINKED ISSUE #1: Create the esn-frontend-inbox-calendar module
Details: https://ci.linagora.com/linagora/lgs/openpaas/esn/-/wikis/Week-by-Week-Activity#esn-frontend-inbox-calendar

### PR DESCRIPTION
**Pull request by tuanlc**
- [x]  Fullcalendar error
- [x] Propose new time modal on blue bar
- [x] Open event detail modal on blue bar

mail_blue-bar

### CODE REVIEW

File: src/app/app.module.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    'esn.calendar',
+    'linagora.esn.resource'
+  ]);
+})(angular);
+
+require('esn-frontend-common-libs/src/frontend/js/modules/session.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/notification.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/i18n/i18n.module.js');
+
+require('esn-frontend-calendar/src/linagora.esn.calendar/app/app.js');
```
Comment (MichaelBailly) [inline src/app/app.module.js]: You are requiring all the calendar application files in inbox. You need, in esn-frontend-calendar, to isolate the files/angular modules that you need to import only them in inbox.

File: assets/index.pug [hunk 2]
```diff
@@ -0,0 +1,12 @@
+doctype html
```
Comment (chamerling) [inline assets/index.pug]: This is not an SPA but a lib, so this file is not needed

**Review by chamerling — CHANGES_REQUESTED** [review]

**Review by chamerling — COMMENTED** [review]
I guess this is not an SPA but a library, what about putting everything in src/lib instead of src/app ?

File: src/app/app.module.js (hunk 1 above)
Comment (tuanlc) [reply src/app/app.module.js]: I split the common part to the libs module

File: assets/index.pug (hunk 2 above)
Comment (tuanlc) [reply assets/index.pug]: Removed

**Comment by tuanlc** [comment]
> I guess this is not an SPA but a library, what about putting everything in src/lib instead of src/app ?

IMO, this is an angular module, so use the "app" term is consistent with other modules. For example,  in `esn-frontend-calendar SPA, `linagora.esn.calendar` is a lib for the SPA.

File: src/app/invitation-message/indicator/invitation-message-indicator.component.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+(function(angular) {
+  'use strict';
+
+  angular.module('esn-frontend-inbox-calendar')
+    .component('calInboxInvitationMessageIndicator', {
+      template: require("./invitation-message-indicator.pug")
+    });
+})(angular);
```
Comment (asmaNode) [inline src/app/invitation-message/indicator/invitation-message-indicator.component.js]: add blank line at the end

File: src/module-info.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "coreModules": [],
+  "dependenceModules": []
+}
```
Comment (asmaNode) [inline src/module-info.json]: add blank line

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  devtool: "source-map",
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+  externals: {
+    angular: 'angular',
+    lodash: 'lodash'
+  }
+};
```
Comment (asmaNode) [inline webpack.config.js]: some here

File: src/app/invitation-message/indicator/invitation-message-indicator.run.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const _ = require('lodash');
```
Comment (chamerling) [inline src/app/invitation-message/indicator/invitation-message-indicator.run.js]: DO you still need dynamic directive?

Comment (tuanlc) [reply src/app/invitation-message/indicator/invitation-message-indicator.run.js]: I understand your point, in my case, the inject condition is quite complicated
```
function runBlock(dynamicDirectiveService, DynamicDirective, INVITATION_MESSAGE_HEADERS) {
    var shouldInject = function(scope) {
      return scope.email &&
        scope.email.headers &&
        INVITATION_MESSAGE_HEADERS.UID in scope.email.headers;
    };
    var directive = new DynamicDirective(shouldInject, 'cal-inbox-invitation-message-blue-bar', {
      attributes: [{ name: 'message', value: 'email' }]
    });

    dynamicDirectiveService.addInjection('inbox-message-info', directive);
```
So, is it ok if I keep it like this for now?

Comment (chamerling) [reply src/app/invitation-message/indicator/invitation-message-indicator.run.js]: I do not get it, didn't we said goodbye to dynamic directives?

Comment (MichaelBailly) [reply src/app/invitation-message/indicator/invitation-message-indicator.run.js]: What Tuan say is that dynamic directives provides a way to choose whether we display, or not, the widget.
Here it's:
```
return scope.email &&
        scope.email.headers &&
        INVITATION_MESSAGE_HEADERS.UID in scope.email.headers;
```
If we ask developers to switch away from dynamic directives, Christophe can you give to the developers the mthod to choose component display ?
Maybe something like: 
```
$scope.mivisible = function() {
      return $scope.email &&
        $scope.email.headers &&
        INVITATION_MESSAGE_HEADERS.UID in $scope.email.headers;
    };
<inbox-message-info ng-if='mevisible()'>
```

Please Christophe provide tech guidance here. Tks

Comment (tuanlc) [reply src/app/invitation-message/indicator/invitation-message-indicator.run.js]: I removed the dynamic directive

File: src/app/app.constants.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+angular.module('esn-frontend-inbox-calendar')
```
Comment (chamerling) [inline src/app/app.constants.js]: We have all our modules named like`esn.something`, can you do the same?

**Review by chamerling — CHANGES_REQUESTED** [review]

File: package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  "dependencies": {
+    "angular": "1.3.20",
+    "esn-frontend-common-libs": "github:openpaas-suite/esn-frontend-common-libs#main",
+    "esn-frontend-calendar": "github:openpaas-suite/esn-frontend-calendar#main",
+    "fullcalendar": "3.9.0",
+    "lodash": "2.4.1"
+  },
+  "devDependencies": {
+    "webpack": "^4.43.0",
+    "webpack-cli": "^3.3.11"
```
Comment (itswillta) [inline package.json]: This is just a intermediary module, so there's no need for `webpack` here right?

File: webpack.config.js (hunk 5 above)
Comment (itswillta) [inline webpack.config.js]: This is just a intermediary module, so there's no need for `webpack` here right?

**Review by itswillta — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (tuanlc) [reply package.json]: Removed

File: webpack.config.js (hunk 5 above)
Comment (tuanlc) [reply webpack.config.js]: Removed this file

File: src/app/app.constants.js (hunk 7 above)
Comment (tuanlc) [reply src/app/app.constants.js]: Renamed

**Review by chamerling — DISMISSED** [review]

**Review by itswillta — DISMISSED** [review]

**Review by itswillta — APPROVED** [review]

**Review by chamerling — APPROVED** [review]

**Pull request merged**